module/workflow/src/Infrastructure/Persistence/Query/DbalStatusQuery.php

```diff
@@ -131,33 +131,30 @@ public function getAllCodes(): array
     /**
      * {@inheritdoc}
      */
-    public function getStatusCount(Language $language): array
+    public function getStatusCount(Language $translationLanguage, Language $workflowLanguage): array
     {
-        $sql = "SELECT id, code, name->>:language AS label FROM status;";
-        $stmt = $this->connection->executeQuery($sql, ['language' => (string) $language]);
-        $statuses = $stmt->fetchAll();
-
-        $sql = 'SELECT vt.value, count(pv.product_id)
-            FROM attribute a
-            JOIN product_value pv ON a.id = pv.attribute_id
-            JOIN value_translation vt ON pv.value_id = vt.value_id
-            WHERE code = :code
-            GROUP BY vt.value
+        $sql = 'SELECT s.code, s.name->>:translationLanguage AS label, s.id AS status_id, count(pws.product_id) AS value
+            FROM status s
+            LEFT JOIN product_workflow_status pws ON s.id = pws.status_id
+            WHERE pws.language = :workflowLanguage
+            GROUP BY s.id, s.code, label
+            UNION
+            SELECT s.code, s.name->>:translationLanguage AS label, s.id, 0 AS value FROM status s
         ';
-        $stmt = $this->connection->executeQuery($sql, ['code' => StatusSystemAttribute::CODE]);
-        $products = $stmt->fetchAll();
+        $stmt = $this->connection->executeQuery(
+            $sql,
+            [
+                'translationLanguage' => (string) $translationLanguage,
+                'workflowLanguage' => (string) $workflowLanguage,
```

Comment:
```suggestion
                'translationLanguage' =>  $translationLanguage->getCode(),
                'workflowLanguage' => $workflowLanguage->getCode(),
```

Reply:
From my perspective, it's just the fact I care only about appropriate string representation for query unless there's any reason that getting "manually" is better?

Reply:
There is no reason